bentoml/_internal/utils/_formparser.py

```diff
@@ -0,0 +1,180 @@
+import binascii
+import os
+import typing as t
+from collections import defaultdict
+
+import multipart.multipart as multipart  # noqa
+from starlette.datastructures import MutableHeaders
+from starlette.formparsers import _user_safe_decode  # noqa
+from starlette.formparsers import Headers, MultiPartMessage
+from starlette.formparsers import MultiPartParser as _StarletteMultiPartParser
+from starlette.requests import Request
+from starlette.responses import Response, StreamingResponse
+
+from ...exceptions import BentoMLException
+
+_ItemsBody = t.TypeVar(
+    "_ItemsBody",
+    bound=t.List[t.Tuple[str, t.List[t.Tuple[bytes, bytes]], bytes]],
+)
+
+
+class MultiPartParser(_StarletteMultiPartParser):
```

Comment:
Relying on external codes is not recommended since they may change without warning. copy/paste is much better.